app/views/games/show.html.erb

```diff
@@ -66,20 +58,33 @@
         var piece = $.parseHTML(
           '<div class="piece" data-id='+ piece.id +'>' + pieceHtml + '</div>');
 
-        $chessboard.find('.square#'+squareId).append(piece);
-        $chessboard.find('.piece').draggable();
+        $('.square#'+squareId).append(piece);
+        $('.piece').draggable();
       }
+    }
 
-      function movePiece(piece) {
-        var coord = piece.closest('.square').attr('id');
-        var newRow = coord[0];
-        var newCol = coord[1];
-
-        $.post(piecesURL + "/" + piece.data("id"), {
-          _method: "PUT",
-          piece: { row: newRow, col: newCol }
-        }).success();
+    $(".square").droppable({
+      drop: function(event, ui) {
+        var coord = $(this).attr('id').split('');
+        var $piece = $(ui.draggable);
+        movePiece($piece, coord);
       }
     });
+
+    // race condition without setTimeout():
+    // The get request may complete before the post request's back-end logic is
+    //   completely executed.
+    // In the worst case, the get request will retrieve pieces before they are
+    //   updated by Piece#move_to!.
```

Comment:
To make complex software processes easy to reason about, there's been a push in the JavaScript world (inspired from everywhere else), to promote a "unidirectional data flow", which is actually what this does here.  It essentially means that the user action (moving a piece) updates the data in some what, then when the data updates, the info displayed to the user is updated.  Something like:

* user move -> Rails server
* Rails server -> HTML board
* HTML board -> user move

So whenever the board updates, the entire board is updated to reflect the current state.  Just like in a GUI app where a user action triggers a change in the application state, then that state is used to update the GUI.  This is a major principle behind ReactJS, if you ever get into that.